sources/api/netdog/src/networkd/config/network.rs

```diff
@@ -0,0 +1,81 @@
+use networkd_derive::{NetworkDFile, NetworkDSection};
+
+#[derive(Debug, Default, NetworkDFile)]
+pub(crate) struct NetworkConfig {
+    match_section: Option<MatchSection>,
+    link_section: Option<LinkSection>,
+    network_section: Option<NetworkSection>,
+    route_sections: Vec<RouteSection>,
+    dhcp4_section: Option<Dhcp4Section>,
+    dhcp6_section: Option<Dhcp6Section>,
+}
```

Comment:
Kind of an aesthetic nit but I find the `_section` suffixes on the fields intrusive:
```suggestion
#[derive(Debug, Default, NetworkDFile)]
pub(crate) struct NetworkConfig {
    match: Option<MatchSection>,
    link: Option<LinkSection>,
    network: Option<NetworkSection>,
    routes: Vec<RouteSection>,
    dhcp4: Option<Dhcp4Section>,
    dhcp6: Option<Dhcp6Section>,
}
```

Reply:
Fair - I waffled on this a bit early on and settled on the `_section` prefixes to be perhaps overly descriptive.  I like them better without as well.